src/main/kotlin/FunCommands.kt

```diff
@@ -0,0 +1,103 @@
+import com.velocitypowered.api.command.SimpleCommand
+import com.velocitypowered.api.proxy.Player
+import com.velocitypowered.api.proxy.ProxyServer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import net.kyori.adventure.text.Component
+import org.slf4j.Logger
+import java.nio.file.Files
+import java.nio.file.Paths
+
+@Serializable
+data class FunCommandConfig(
+    val command: String,
+    val description: String,
+    val globalChat: String,
+    val run: String? = null // Optional execution logic
+)
+
+class FunCommands(
+    private val proxy: ProxyServer,
+    private val logger: Logger,
+    private val chattORE: ChattORE
+) {
+
+    fun loadFunCommands() {
+        val resourcePath = "commands.json"
+        val resourceStream = javaClass.classLoader.getResourceAsStream(resourcePath)
+
+        if (resourceStream == null) {
+            logger.warn("$resourcePath not found. Skipping fun command loading.")
+            return
+        }
+
+        val commandsJson = resourceStream.bufferedReader().use { it.readText() }
+        val commands = Json.decodeFromString<List<FunCommandConfig>>(commandsJson)
+        logger.info("Parsed ${commands.size} commands from JSON.")
+
+        commands.forEach { commandConfig ->
+            val meta = proxy.commandManager.metaBuilder(commandConfig.command).build()
+            proxy.commandManager.register(meta, createDynamicCommand(commandConfig))
+        }
+
+        logger.info("Loaded ${commands.size} fun commands.")
+    }
+
+    private fun createDynamicCommand(commandConfig: FunCommandConfig): SimpleCommand {
+        return SimpleCommand { invocation ->
+            val source = invocation.source()
+            val args = invocation.arguments()
+
+            if (source !is Player) {
+                source.sendMessage(Component.text("This command can only be used by players!"))
+                return@SimpleCommand
+            }
+
+            val player = source
+            val message = commandConfig.globalChat
+                .replace("<name>", player.username)
+                .replace("<arg-all>", args.joinToString(" "))
+                .replace("$1", args.getOrNull(1) ?: "<missing>")
+                .replace("$2", args.getOrNull(2) ?: "<missing>")
+
+
+            chattORE.broadcast(
+                message.render(
+                    mapOf(
+                        "message" to message.toComponent(),
+                        "sender" to player.username.toComponent()
+                    )
+                )
+            )
+
+
+
+            chattORE.logger.info(
+                "Executed command: /${commandConfig.command} by ${player.username} with args: ${
+                    args.joinToString(
+                        " "
+                    )
+                }"
+            )
+
+            commandConfig.run?.let { executeAction(it, player) }
+        }
+    }
+
+    private fun executeAction(action: String, player: Player) {
```

Comment:
There *may* be a better way to do this, but I like the idea of having an "allow list" of what command actions can be performed.

However, we may have to reconsider that if we turn this into an aliasing system, but maybe a separate `aliases.json` for that would be best anyway.